pkg/cmd/cluster/connect/connect.go

```diff
@@ -70,43 +76,64 @@ func runConnect(opts *options) error {
 		return err
 	}
 
-	clusterConn, err := cluster.NewKubernetesClusterConnection(conn, opts.Config, opts.Logger, opts.kubeconfigLocation, opts.IO, opts.localizer)
+	conn.API()
+
+	err = validateServiceID(opts.Context, conn.API(), opts.serviceType, opts.serviceID)
 	if err != nil {
 		return err
 	}
 
-	cfg, err := opts.Config.Load()
+	clusterConn, err := cluster.NewKubernetesClusterConnection(conn, opts.Config, opts.Logger, opts.kubeconfigLocation, opts.IO, opts.localizer)
 	if err != nil {
 		return err
 	}
 
-	// In future config will include Id's of other services
-	if cfg.Services.Kafka == nil || opts.ignoreContext {
-		// nolint
-		selectedKafka, err := kafka.InteractiveSelect(opts.Context, conn, opts.Logger, opts.localizer)
-		if err != nil {
-			return err
-		}
-		if selectedKafka == nil {
-			return nil
-		}
-		opts.selectedKafka = selectedKafka.GetId()
-	} else {
-		opts.selectedKafka = cfg.Services.Kafka.ClusterID
-	}
-
 	arguments := &cluster.ConnectArguments{
 		OfflineAccessToken:      opts.offlineAccessToken,
 		ForceCreationWithoutAsk: opts.forceCreationWithoutAsk,
-		IgnoreContext:           opts.ignoreContext,
-		SelectedKafka:           opts.selectedKafka,
 		Namespace:               opts.namespace,
+		SelectedService:         opts.serviceType,
+		SelectedServiceID:       opts.serviceID,
 	}
 
-	err = clusterConn.Connect(opts.Context, arguments)
+	connectOpts := cluster.Options{
+		IO:         opts.IO,
+		Logger:     opts.Logger,
+		Localizer:  opts.localizer,
+		Config:     opts.Config,
+		Connection: conn,
+	}
+
+	var service cluster.CustomConnection
+
+	switch opts.serviceType {
```

Comment:
[ConstantsLeak]

There is multiple places where we iterate over Kafka/Registry types. It will be better if we do it once and create instances of those objects. 
Then instances can be used for validation etc.

This is leaking logic out of the abstraction

Reply:
Every time we iterate over types 💡  should appear with `Should We use Strategy pattern here`?